alacritty_terminal/src/input.rs

```diff
@@ -445,50 +484,45 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         }
     }
 
-    /// Underline URLs and change the mouse cursor when URL hover state changes.
-    fn update_url_highlight(&mut self, point: Point, modifiers: ModifiersState) {
-        let mouse_mode =
-            TermMode::MOUSE_MOTION | TermMode::MOUSE_DRAG | TermMode::MOUSE_REPORT_CLICK;
+    fn url_bounds_at_point(&self, url: Url, point: Point) -> RangeInclusive<Linear> {
```

Comment:
This method is a bit messy I think and `url_bounds_at_point` might be a little hard to understand, however I think with the rfind_url changes I've made it should be possible to simplify this down to something more straight-forward like `url_bounds(url) -> Range<Linear>` which would make it pretty clear.

So I think this is good to go for now and I'd just worry about potential improvements to this with the other PR which I'd like to land after this one.